### PR TITLE
Remove GVRScript; PostEffectShader changes to support removal of GVRScript

### PR DESCRIPTION
**Pull request by liaxim**
Demos pull request: https://github.com/gearvrf/GearVRf-Demos/pull/339

GearVRf-DCO-1.0-Signed-off-by: Mihail Marinov m.marinov@samsung.com

### CODE REVIEW

**Comment by liaxim** [comment]
Ready for review

**Comment by thomasflynn** [comment]
this looks like more than just the PostEffectShader changes.  it also has all the code to remove GVRScript.  was that intended?

**Comment by liaxim** [comment]
Yes, intended. Should I split the change in two?

**Comment by thomasflynn** [comment]
I think so.  I'd be fine with removing GVRScript in the next release.  For now, I want to keep things stable until we get a few more bugs fixed and then do a release.

**Comment by liaxim** [comment]
The change to the PostEffectShader can wait too I think. We could have it in after the release. It is a prerequisite to the removal of GVRScript more than anything.

**Comment by NolaDonato** [comment]
My patch that removes C++ builtin shaders makes some changes to the PostEffectShader stuff too.

**Comment by liaxim** [comment]
@NolaDonato If you patch for removing the builtin shaders will come soon enough then this pull request will wait for it.

**Comment by NolaDonato** [comment]
Don’t wait. They are now merging my patch with uniform buffers. This will take some time.

**Comment by liaxim** [comment]
This is ready for review, testing and merging eventually.

No problems with the samples have been found. Works equally well for all backends we support.

All the automated framework tests that do not involve image comparison pass.

**Comment by liaxim** [comment]
@rahul27 Do you think you could have a look at this change? Thanks.

**Comment by rahul27** [comment]
@liaxim looking at it now.

**Comment by rahul27** [comment]
@liaxim are you able to build the extensions using command line?

**Comment by liaxim** [comment]
Yes, the extensions build for me from the command line.

**Comment by rahul27** [comment]
Do you use `./gradlew assembleDebug` ?

I get `Observed package id 'ndk-bundle' in inconsistent location `.

**Comment by liaxim** [comment]
Yes. This is the one. There is something off with your local Android SDK installation. Usually it is not fatal.. the build machine has been getting these forever.

**Comment by rahul27** [comment]
gvr-jassimp and meshanimation crash on launch for me with this change.

**Comment by liaxim** [comment]
Do you have stacktraces?

**Comment by liaxim** [comment]
Tried on Note5 and S7 - worked for me. ``git clean -fdx`` might be advisable.

**Comment by rahul27** [comment]
Fixed the command line build for the extensions. I had the wrong path to ANDROID_HOME in my bash file. It now builds correctly.

Reg. gvr-jassimp it still crashes. I don't see a visible crash trace on logcat.

**Comment by rahul27** [comment]
@liaxim gvr-jassimp works when the wifi is on but crashes when it is off. 

This does not happen on master.

**Comment by liaxim** [comment]
gvr-jassimp: made a small tweak to the app itself.

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GVRActivity.java [hunk 1]
```diff
@@ -240,10 +240,10 @@ protected void onDestroy() {
     }
 
     /**
-     * Links {@linkplain GVRScript a script} to the activity; sets the version;
+     * Links {@linkplain GVRMain a script} to the activity; sets the version;
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GVRActivity.java]: Good point. Will make the changes.

**Comment by liaxim** [comment]
Will merge tomorrow after lunch

**Pull request merged**